Suggest columns for `ORDER BY` and `DISTINCT` (fixes #685)

owst opened the pull request:
## Description
Having typed an alias name in an `ORDER BY` or (`SELECT`) `DISTINCT` clause,
the alias was not taken account of, and the completion simply listed all
columns. This change fixes that, and makes the autocompletion behave the
same as in `SELECT` and `WHERE` clauses.

## Checklist
<!--- We appreciate your help and want to give you credit. Please take a moment to put an `x` in the boxes below as you complete them. -->
- [x] I've added this contribution to the `changelog.md`.
- [x] I've added my name to the `AUTHORS` file (or it's already there).

amjith commented:
Thanks for taking the time to make pgcli better. Sorry I haven't had time to look into this. I won't have time till end of this week, but if none of the other core devs have reviewed it by then, I'll take a look. Sorry about the delay.

j-bennet commented on tests/test_sqlcompletion.py:
```diff
@@ -4,10 +4,17 @@
 
-# Returns the expected select-clause suggestions for a single-table select
+
+def table_reference(table, schema=None, alias=None, is_function=False):
```
`TableReference` instantiation is already a one-liner. What's the point of wrapping it into a function?

owst replied:
On reflection, probably none, I think I added it before I did a bunch of line-wrapping. Of course, one can avoid passing a few parameters since `table_reference` has default parameters, but I don't think the method pulls its weight, I'll remove it.

j-bennet commented on tests/test_sqlcompletion.py:
```diff
@@ -213,9 +220,62 @@ def test_truncate_suggests_qualified_tables():
 def test_distinct_suggests_cols(text):
     suggestions = suggest_type(text, text)
-    assert suggestions ==(Column(table_refs=(), qualifiable=True),)
+    assert set(suggestions) == set([
```
Well done adding tests!

owst replied:
Wouldn't be a good PR without them...!

owst commented:
No worries about taking your time @amjith! Thanks for the review @j-bennet - I've addressed the comment and squashed it into one commit since it was a small tweak (I hope that's ok for this project!)

j-bennet commented:
@owst Also, the line in changelog:

 "Fixed completions after ORDER BY and DISTINCT when using table aliases".

owst commented:
Yeah, fair enough, I'm certainly not wedded to the current wording. The only thing is "fixed" to me suggests it was "broken" before, when I think it was just a missing feature. Perhaps 
> Completions after ORDER BY and DISTINCT now take account of table aliases.

?

j-bennet commented:
@owst Yes, that is better!

j-bennet commented:
Looks good. Merging.

Thank you!

🌷
